docs/style-guide.md

```diff
@@ -0,0 +1,291 @@
+# Rust Style Guide
+
+## Motivation - why not to completely rely a formatting tool?
```

Comment:
This question conflates formatting with a style. Style is so much more, as next paragraph answers. 
But it is all pretty obvious. I'd state the question — "why not rely on a formatting tool for formatting". And that second paragraph answers.